src/targets/gpu/jit/ck_gemm_softmax_gemm.cpp

```diff
@@ -134,14 +135,19 @@ struct ck_gemm_softmax_gemm_compiler : compiler<ck_gemm_softmax_gemm_compiler>
         auto batch_count    = get_batch_count(c_shape);
         auto problem        = create_problem(inputs, v);
 
-        const auto include_header   = problem.GetIncludeHeader();
-        const auto solutions        = problem.GetSolutions(ctx.get_current_device().get_gfx_name());
+        const auto include_header = problem.GetIncludeHeader();
+        const auto solutions =
+            problem.GetSolutions(ctx.get_current_device().get_gfx_name(), "", "");
         const auto& solution        = solutions.at(tuning_value);
-        const auto template_str     = solution.template_str;
-        const auto blocks_per_batch = solution.grid_size;
-        const auto block_size       = solution.block_size;
+        const auto template_str     = solution.ToTemplateString();
+        const auto block_size       = solution.GetTemplateParameter<std::size_t>("BlockSize");
+        const auto m_per_block      = solution.GetTemplateParameter<std::size_t>("Gemm01MPerBlock");
+        const auto n1_per_block     = solution.GetTemplateParameter<std::size_t>("Gemm1NPerBlock");
+        const auto blocks_per_batch = ck::host::integer_divide_ceil(problem.M, m_per_block) *
+                                      ck::host::integer_divide_ceil(problem.O, n1_per_block);
 
         hip_compile_options options;
+        options.emplace_param("-DCK_CODE_GEN_RTC");
```

Comment:
What is that flag supposed to do, and whats the difference between this and `__HIPCC_RTC__`?

Reply:
This flag was introduced by the CK team to enable hipRTC with MIOpen. We are taking advantage of that here. It essentially serves the same purpose as `__HIPCC_RTC__`. I'm not sure why they didn't use `__HIPCC_RTC__` for everything.